dts/bindings/timer/atmel,sam0-tc32.yaml

```diff
@@ -26,3 +26,15 @@ properties:
 
   label:
     category: required
+
+  clock-frequency:
```

Comment:
Why isn't the clock-freq figured out from the clock for this device?

Reply:
Two reasons:

1. The TC peripheral itself has an on-peripheral divisor available, so this is used to calculate what that divisor should be (similar to how the SPI drivers calculate their clock rate).
2. The `counter` API requires that the frequency be available at compile time (`.info`), so we can't use the `clock_control` API to query it (since that's on available at run time).

I have clarified the description on the binding.